HTTPProxy: add healthCheck port config (#4761)
projectcontour/contour: changelogs/unreleased/4761-yangyy93-minor.md

## Add health check port for HTTP health check and TCP health check

> HTTPProxy.Route.Service and HTTPProxy.TCPProxy.Service now has a `HealthPort` which specifies a health check port that is different from the routing port.If the field was not specified,it is the same as `Port`.

Reviewer: ```suggestion
HTTPProxy.Route.Service and HTTPProxy.TCPProxy.Service now has a `HealthPort` field which specifies a health check port that is different from the routing port. If not specified, the service `Port` field is used for healthchecking.
```
